org.eclipse.xtext/src/org/eclipse/xtext/build/Indexer.java

```diff
@@ -198,10 +198,13 @@ public URI getURI() {
 				.transform(IResourceDescription::getURI).copyInto(new HashSet<>());
 		remainingURIs.removeAll(FluentIterable.from(deltas).transform(Delta::getUri).toSet());
 		List<URI> allAffected = FluentIterable.from(remainingURIs).filter(it -> {
-			IResourceDescription.Manager manager = context.getResourceServiceProvider(it)
-					.getResourceDescriptionManager();
-			IResourceDescription resourceDescription = previousIndex.getResourceDescription(it);
-			return isAffected(resourceDescription, manager, allDeltas, allDeltas, newIndex);
+			IResourceServiceProvider resourceServiceProvider = context.getResourceServiceProvider(it);
```

Comment:
how do you end up with non rsp resources in remainingURIs

Reply:
I can imagine a persisted index that is being reused after some language was deinstalled.
In any case, this should at least mark the URI for removal.

Reply:
Thanks, I have done that as, now, the URI will  be marked for removal. I also left two comments for two existing null checks in the code, which I think could be removed, in case you want to take a look at them.

Reply:
@szarnekow , I have added code to mark the URI for removal. Is the code now good for you? Thanks